src/tests/pmfs_demo.c

```diff
@@ -0,0 +1,1528 @@
+/**
+ * (C) Copyright 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+
+#include <stdio.h>
+#include <fcntl.h>
+#include <string.h>
+#include <errno.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+#include <sys/ioctl.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <getopt.h>
+#include <daos/common.h>
+#include <daos/tests_lib.h>
+#include <daos_srv/vos.h>
+#include <pmfs/pmfs.h>
+#include <pmfs/vos_target_fs.h>
+#include <pmfs/vos_tasks.h>
+#include "perf_internal.h"
+static struct vos_fs_cmd_args *g_vfca;
+static int count;
+static int num;
+D_LIST_HEAD(g_test_pool_list);
+D_LIST_HEAD(g_test_fini_list);
+D_LIST_HEAD(g_present_list);
+
+enum {
+	MKFS,
+	MOUNT
+};
+
+struct mkfs_args {
+	daos_handle_t poh;
+	uuid_t uuid;
+};
+
+struct mount_args {
+	daos_handle_t poh;
+	daos_handle_t coh;
+	int flags;
+	struct pmfs **pmfs;
+};
+
+struct umount_args {
+	struct pmfs *pmfs;
+};
+
+struct mkdir_args {
+	struct pmfs *pmfs;
+	struct pmfs_obj *parent;
+	const char *name;
+	mode_t mode;
+};
+
+struct listdir_args {
+	struct pmfs *pmfs;
+	struct pmfs_obj *obj;
+	uint32_t nr;
+};
+
+struct remove_args {
+	struct pmfs *pmfs;
+	struct pmfs_obj *parent;
+	const char *name;
+	bool force;
+	daos_obj_id_t *oid;
+};
+
+struct open_args {
+	struct pmfs *pmfs;
+	struct pmfs_obj *parent;
+	const char *name;
+	mode_t mode;
+	int flags;
+	daos_size_t chunk_size;
+	const char *value;
+	struct pmfs_obj *obj;
+};
+
+struct readdir_args {
+	struct pmfs *pmfs;
+	struct pmfs_obj *obj;
+	uint32_t *nr;
+	struct dirent *dirs;
+};
+
+struct lookup_args {
+	struct pmfs *pmfs;
+	const char *path;
+	int flags;
+	struct pmfs_obj *obj;
+	mode_t *mode;
+	struct stat *stbuf;
+};
+
+struct release_args {
+	struct pmfs_obj *obj;
+};
+
+struct punch_args {
+	struct pmfs *pmfs;
+	struct pmfs_obj *obj;
+	daos_off_t offset;
+	daos_size_t len;
+};
+
+struct write_args {
+	struct pmfs *pmfs;
+	struct pmfs_obj *obj;
+	d_sg_list_t *user_sgl;
+	daos_size_t buf_size;
+	daos_off_t off;
+	daos_size_t *write_size;
+};
+
+struct read_args {
+	struct pmfs *pmfs;
+	struct pmfs_obj *obj;
+	d_sg_list_t *user_sgl;
+	daos_off_t off;
+	daos_size_t stride_size;
+	daos_size_t *read_size;
+};
+
+struct stat_args {
+	struct pmfs *pmfs;
+	struct pmfs_obj *parent;
+	const char *name;
+	struct stat *stbuf;
+};
+
+static void
+pmfs_buffer_render(char *buf, unsigned int buf_len)
+{
+	int     nr = 'z' - 'a' + 1;
+	int     i;
+
+	for (i = 0; i < buf_len - 1; i++) {
+		int randv = rand() % (2 * nr);
+
+		if (randv < nr)
+			buf[i] = 'a' + randv;
+		else
+			buf[i] = 'A' + (randv - nr);
+	}
+	buf[i] = '\0';
+}
+
+static  struct pmfs_pool
+pmfs_add_single_pool(char *tsc_pmem_file, uint64_t tsc_nvme_size,
+	 	     uint64_t tsc_scm_size, bool tsc_skip_cont_create,
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs